### PR TITLE
Anshu-Assignment-22-10min+10min-Done

### PR DESCRIPTION
**Pull request by AnshuKJun20**
image
image

### CODE REVIEW

File: src/anshuKJun20/PalindromeNumber.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+			add = add * 10 + remainder;
+		}
+		if (add == number1) {
+			System.out.println("Given number " + number1 + " is palindrome.");
+		} else {
+			System.out.println("Given number " + number1 + " is not palindrome.");
+		}
+	}
+
+	public void display() {
```
Comment (PranitaPJun20) [inline src/anshuKJun20/PalindromeNumber.java]: Here no need of extra method.You can directly print details by calling core method.

File: src/anshuKJun20/PalindromeNumber.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package anshuKJun20;
+
+public class PalindromeNumber {
+	private void findPalindromeNumber(int number) {
+		int number1 = number, add = 0;
+		while (number > 0) {
+			int remainder = number % 10;
+			number = number / 10;
+			add = add * 10 + remainder;
```
Comment (PranitaPJun20) [inline src/anshuKJun20/PalindromeNumber.java]: You can optimize the code by merging line 7 in line 9.
i.e. At line 9 write add = (add *10)+(number%10);
Remove line 7

File: src/anshuKJun20/StringPalindrome.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+				result = result + name.charAt(index);
+			}
+			if (name.equalsIgnoreCase(result)) {
+				System.out.println("Given string " + name + " is palindrome");
+			} else {
+				System.out.println("Given string " + name + " is not palindrome");
+			}
+		}
+
+		public void display() {
```
Comment (PranitaPJun20) [inline src/anshuKJun20/StringPalindrome.java]: You are using display method to call core method.You can use it.But i think in this simple program we don't need this. @KrishnaKTechnocredits  Please correct me if I am wrong.

**Review by PranitaPJun20 — COMMENTED** [review]
Please work on corrections.
We can clear display() method point from Krishna.

File: src/anshuKJun20/PalindromeNumber.java (hunk 1 above)
Comment (AnshuKJun20) [reply src/anshuKJun20/PalindromeNumber.java]: chaanges done.

File: src/anshuKJun20/StringPalindrome.java (hunk 3 above)
Comment (AnshuKJun20) [reply src/anshuKJun20/StringPalindrome.java]: yes Pranita you are right. I have  made changes according to this.

File: src/anshuKJun20/PalindromeNumber.java (hunk 2 above)
Comment (AnshuKJun20) [reply src/anshuKJun20/PalindromeNumber.java]: Changes done. Thankyou for reviewing it.

**Review by PranitaPJun20 — APPROVED** [review]
Corrections done as per expectation.

File: src/anshuKJun20/StringPalindrome.java (hunk 3 above)
Comment (KrishnaKTechnocredits) [reply src/anshuKJun20/StringPalindrome.java]: Yes correct @PranitaPJun20 we don't need display() method.

File: src/anshuKJun20/PalindromeNumber.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package anshuKJun20;
+
+public class PalindromeNumber {
+	void findPalindromeNumber(int number) {
+		int number1 = number, add = 0;
+		while (number > 0) {
+			add = add * 10 + number % 10;
```
Comment (KrishnaKTechnocredits) [inline src/anshuKJun20/PalindromeNumber.java:7]: Use revNumber instead of add.

File: src/anshuKJun20/StringPalindrome.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package anshuKJun20;
+
+public class StringPalindrome {
+
+	void finfPalindromeString(String name) {
+		String result = "";
+		for (int index = (name.length() - 1); index >= 0; index--) {
+			result = result + name.charAt(index);
```
Comment (KrishnaKTechnocredits) [inline src/anshuKJun20/StringPalindrome.java:8]: revString instead of result.

File: src/anshuKJun20/StringPalindrome.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+		}
+		if (name.equalsIgnoreCase(result)) {
```
Comment (KrishnaKTechnocredits) [inline src/anshuKJun20/StringPalindrome.java:10]: Use .equals() method only for palindrome. Because Radar is not Palindrome string.

**Pull request merged**